src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Windows", "Linux")]
+        public string OSType { get; set; }
+        
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = true, Position = 3, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string[] DiskLink { get; set; }
+        
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = true, Position = 3, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string[] DiskFile { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 4, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 4, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string VMName { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 5, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 5, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string VMSize { get; set; }
+        
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 6, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 6, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public List<SecurityRule> SecurityRules { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 7, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 7, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string DefaultVNetAddressSpace { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 8, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 8, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string DefaultSubnetAddressSpace { get; set; }
+
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = false, Position = 9, ValueFromPipelineByPropertyName = true)]
+        [Parameter(ParameterSetName = DiskFileParameterSetNameStr, Mandatory = false, Position = 9, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [ValidateRange(1, 64)]
+        public int? NumberOfUploaderThreads { get; set; }
```

Comment:
use "int" instead of "int?"

Reply:
Then I need to introduce one more ```int``` member variable to hold the real value. It doesn't seem to be really worth it...

Reply:
@huangpf it is worth it to follow cmdlet standards, so that, if I have a nullable value type, there is actually some reason to assign a nullable value

Reply:
OK